requests_kerberos/kerberos_.py

```diff
@@ -117,9 +120,13 @@ def generate_request_header(self, response, host, is_preemptive=False):
             # w/ name-based HTTP hosting)
             kerb_host = self.hostname_override if self.hostname_override is not None else host
             kerb_spn = "{0}@{1}".format(self.service, kerb_host)
+            
+            kwargs = {'principal': self.principal}
+            if self._using_kerberos_sspi and self.principal:
+                raise NotImplementedError("Can't use 'principal' argument with kerberos-sspi.")
```

Comment:
kerberos-sspi doesn't recognize the `principal` keyword argument at all, so it can't even take `None`.